runtime/compiler/p/codegen/PPCRecompilation.cpp

```diff
@@ -135,9 +135,9 @@ TR::Instruction *TR_PPCRecompilation::generatePrologue(TR::Instruction *cursor)
          {
          intptr_t adjustedAddr = HI_VALUE(addr);
          // lis gr11, upper 16-bits
-         cursor = generateTrg1ImmInstruction(cg(), TR::InstOpCode::lis, firstNode, gr11, (int16_t)adjustedAddr>>32, cursor );
+         cursor = generateTrg1ImmInstruction(cg(), TR::InstOpCode::lis, firstNode, gr11, adjustedAddr >> 32, cursor );
```

Comment:
I'm not sure what the original code was expecting.  Was it written thinking that shifting had higher precedence than the C-style cast?  If so, and `adjustedAddr` was 0x800180028003, say, then I think the expectation was that the shift would yield an `intptr_t` value of 0x0000000000008001 and then the cast would result in the `int16_t` 0x8001, which passed to the `int32_t` parameter of `generateTrg1ImmInstruction` would be promoted to 0xFFFF8001.

But I think the cast actually applies first, yielding an `int16_t` value of 0x8003.  Then I'm not sure what the 32-bit shift would yield - perhaps 0xFFFF which when promoted to `int32_t` for the corresponding parameter would yield 0xFFFFFFFF.

I think the modified code will yield a result of 0xFFFF8001 for my example, which I'm guessing was what was expected.
 
Jimmy @IBMJimmyk, may I ask you to weigh in on this?  I see you had changed this line last in pull request #9866.  Should it simply be changed to this?  And my apologies for my confusion!

```
          cursor = generateTrg1ImmInstruction(cg(), TR::InstOpCode::lis, firstNode, gr11, (int16_t)(adjustedAddr>>32), cursor );
```

Reply:
I think I forgot the parentheses there. It should be `(int16_t)(adjustedAddr>>32)`. I think how it is currently written is undefined behavior. Starting from `0x800180028003`, the `lis` would want an immediate value of `0x8001`.